File: Atcoder_#356/javascript/b_Nutrients.js
function Main(input) {
	const lines = input.split("\n");
  const [n, m] = lines[0].split(" ").map(Number);
  const a = lines[1].split(" ").map(Number);

  let totalNutrients = Array(m).fill(0);
  for (let i = 0; i < n; i++) {
    const x = lines[2 + i].split(' ').map(Number);
    for (let j = 0; j < m; j++) {
      totalNutrients[j] += x[j];
    }
  }

  let achieved = true;
  for (let i = 0; i < m; i++) {
    if (totalNutrients[i] < a[i]) {
      achieved = false;
      break;
    }
  }

  console.log(achieved ? "Yes" : "No");
}
Main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: Atcoder_#356/ruby/b_nutrients.rb
n, m = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

total_nutrients = Array.new(m, 0)

n.times do
  x = gets.split.map(&:to_i)
  m.times do |j|
    total_nutrients[j] += x[j]
  end
end

achieved = true
m.times do |i|
  if total_nutrients[i] < a[i]
    achieved = false
    break
  end
end

if achieved
  puts "Yes"
else
  puts "No"
end
